drivers/weather/aagsolo.cpp

```diff
@@ -0,0 +1,388 @@
+        {
+            minOK = 20;
+            maxOK = 95;
+            percWarn = 15;
+            IUGetConfigNumber(getDeviceName(), "WEATHER_HUMIDITY", "MIN_OK", &minOK);
+            IUGetConfigNumber(getDeviceName(), "WEATHER_HUMIDITY", "MAX_OK", &maxOK);
+            IUGetConfigNumber(getDeviceName(), "WEATHER_HUMIDITY", "PERC_WARN", &percWarn);
+
+            addParameter("WEATHER_HUMIDITY", "Humidity", minOK, maxOK, percWarn);
+            if (criticalSP[5].getState())
+                setCriticalParameter("WEATHER_HUMIDITY");
+        }
+        else if (x.first == keys[RELPRESS].key)
+        {
+            minOK = 983;
+            maxOK = 1043;
+            percWarn = 15;
+            IUGetConfigNumber(getDeviceName(), "WEATHER_PRESSURE", "MIN_OK", &minOK);
+            IUGetConfigNumber(getDeviceName(), "WEATHER_PRESSURE", "MAX_OK", &maxOK);
+            IUGetConfigNumber(getDeviceName(), "WEATHER_PRESSURE", "PERC_WARN", &percWarn);
+
+            addParameter("WEATHER_PRESSURE", "rel. pressure", minOK, maxOK, percWarn);
+            if (criticalSP[6].getState())
+                setCriticalParameter("WEATHER_PRESSURE");
+        }
+        else if (x.first == keys[SAFE].key)
+        {
+            minOK = 0.5;
+            maxOK = 1.5;
+            percWarn = 0;
+            IUGetConfigNumber(getDeviceName(), "WEATHER_ISSAFE", "MIN_OK", &minOK);
+            IUGetConfigNumber(getDeviceName(), "WEATHER_ISSAFE", "MAX_OK", &maxOK);
+            IUGetConfigNumber(getDeviceName(), "WEATHER_ISSAFE", "PERC_WARN", &percWarn);
+
+            addParameter("WEATHER_ISSAFE", "Safe", minOK, maxOK, percWarn);
+            if (criticalSP[8].getState())
+                setCriticalParameter("WEATHER_ISSAFE");
+        }
+    }
+    return true;
+}
+
+// initialize properties on launching indiserver
+bool AAGSolo::initProperties()
+{
+    INDI::Weather::initProperties();
+    addDebugControl();
+    // Critical Parameters --------------------------------------------------------------------
+    criticalSP[0].fill("CRITICAL_1", "Rain", ISS_ON);
+    criticalSP[1].fill("CRITICAL_2", "Temperature", ISS_OFF);
+    criticalSP[2].fill("CRITICAL_3", "Wind", ISS_ON);
+    criticalSP[3].fill("CRITICAL_4", "Gust", ISS_ON);
+    criticalSP[4].fill("CRITICAL_5", "Clouds", ISS_ON);
+    criticalSP[5].fill("CRITICAL_6", "Humidity", ISS_OFF);
+    criticalSP[6].fill("CRITICAL_7", "Pressure", ISS_OFF);
+    criticalSP[7].fill("CRITICAL_8", "Light", ISS_OFF);
+    criticalSP[8].fill("CRITICAL_9", "Safe", ISS_OFF);
+    criticalSP.fill(getDeviceName(), "CRITICALS", "Criticals", OPTIONS_TAB, IP_RW, ISR_NOFMANY, 0, IPS_IDLE);
+    defineProperty(criticalSP);
+    loadConfig(true, "CRITICALS");
+
+    soloInfoTP[0].fill("CWINFO", "cwinfo", nullptr);
+    soloInfoTP[1].fill("DATATIME", "GMT Time", nullptr);
+    soloInfoTP.fill(getDefaultName(), "DEVICEINFO", "Device Info", INFO_TAB, IP_RO, 60, IPS_IDLE);
+    defineProperty(soloInfoTP);
```

Comment:
no defineProperty should be called in initProperties. It should be called in updateProperties.

Also, for:

```criticalSP.fill(getDeviceName(), "CRITICALS", "Criticals", OPTIONS_TAB, IP_RW, ISR_NOFMANY, 0, IPS_IDLE);```

You can simply add after that ```critialSP.load()``` and it would load the config, no need to explicitly call load config.